File: Go/File/file-extension.go
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
	// Get the file extension of a file
	fmt.Println(getFileExtension("assets/valid/valid-txt.txt"))
}

// Get the file extension of a file
func getFileExtension(path string) string {
	return filepath.Ext(path)
}

File: TypeScript/File/file-extension.ts
import * as path from "path"

function main(): void {
    // Get the file extension of a file and return it
    console.log(getFileExtension("assets/valid/valid-json.json"))
}

main()

// Get the file extension of a file and return it
function getFileExtension(filePath: string): string {
    return path.extname(filePath)
}